libcontainer/sync_unix.go

```diff
@@ -26,9 +29,23 @@ func (s *syncSocket) File() *os.File {
 }
 
 func (s *syncSocket) Close() error {
+	s.m.Lock()
+	defer s.m.Unlock()
+
+	// We can set closed to true here anyway without checking the error.
+	s.closed = true
 	return s.f.Close()
 }
 
+func (s *syncSocket) IsNomalClosed() bool {
+	s.m.Lock()
+	defer s.m.Unlock()
+
+	// We need to know that whether the sync socket is normally closed or not.
+	// If the sync socket is closed by accident, we don't care about it.
+	return s.closed
+}
+
```

Comment:
```suggestion
	// Even with errors from Close(), we have to assume the pipe was closed.
	s.closed.Store(true)
	return s.f.Close()
}

func (s *syncSocket) isClosed() bool {
	return s.closed.Load()
}

```

Reply:
I think the name `isClosed` here has some confuse for users, because we don't need to detect whether the socket is really closed or not. For example, if the socket fd is wrongly closed by some other code logic. We just only care about it is closed explicitly by us or not.

Reply:
I mean, the only way we can check if it is closed is if we intentionally closed it. There is no foolproof way on Linux to check if a file descriptor was closed (and a different file descriptor didn't replace it).

Also, to me `isClosed` also sounds like "was `Close` called on this handle" in this context. But if you really want to call it something else, call it something like `isFinished`. The main reasons for including the name change in my suggestion was that I didn't see why it should be exported and `IsNormalClosed` sounds very weird to my ears.